app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -14,15 +14,28 @@ def sync_monitor
   end
 
   def sync_from_system
+    # All miq_server instances have to reside on the same Kubernetes cluster, so
+    # we only have to sync the list of pods and deployments once
+    return unless my_server.is_master?
+
     ensure_kube_monitors_started
-    cleanup_orphaned_worker_rows
-    sync_deployment_settings
   end
 
   def enough_resource_to_start_worker?(_worker_class)
     true
   end
 
+  def sync_workers
+    # Before syncing the workers check for any orphaned worker rows that don't have
+    # a current pod and delete them
+    cleanup_orphaned_worker_rows
+
+    # Update worker deployments with updated settings such as cpu/memory limits
+    sync_deployment_settings
```

Comment:
TODO this seems like something that sync_workers should handle when looping through the worker classes

Reply:
Yes although, we'd be trading guard clauses:

```ruby
# currently we loop through `podified_miq_workers` and skip duplicate deployment names we already processed:

next if checked_deployments.include?(worker.worker_deployment_name)
```

Instead, we'd have to skip worker classes that have no existing `podified_miq_workers` / deployment.